src/widgetastic_patternfly/__init__.py

```diff
@@ -1849,3 +1850,172 @@ def click_location(self, name, handle_alert=True):
+                    self._dates.update({int(el.text): el})
+            return self._dates
+
+    @View.nested
+    class month_pick(HeaderView):   # noqa
+        MONTHS = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+        _months = {}
+
+        @property
+        def _elements(self):
+            for el in self.browser.elements(self.MONTHS):
+                if not bool({'disabled'} & self.browser.classes(el)):
+                    self._months.update({el.text: el})
+            return self._months
+
+    @View.nested
+    class year_pick(HeaderView):    # noqa
+        YEARS = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+        _years = {}
+
+        @property
+        def _elements(self):
+            for el in self.browser.elements(self.YEARS):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    self._years.update({int(el.text): el})
+            return self._years
+
+        def select(self, value):
+            for yr in range(0, 10):
+                start_yr, end_yr = [int(item) for item in self.dp_switch.read().split('-')]
+                if start_yr <= value <= end_yr:
+                    for el, web_el in self._elements.items():
+                        if el == value:
+                            web_el.click()
+                            return True
+                elif value < start_yr:
+                    self.prev_button.click()
+                elif value > end_yr:
+                    self.next_button.click()
+            else:
+                raise ValueError("Not valid year or with more than century difference")
+
+    @property
+    def date_mapp(self):
+        if self.date_format:
+            date_fmt = self.date_format
+            for item in self.date_format.split('/'):
+                date_fmt = date_fmt.replace(item, self.DATE_MAPPING[item])
+            return date_fmt
+        else:
+            return None
+
+    def read(self):
+        """read the selected date
+        Returns:
+            :py:class:`datetime.datetime` or `str`
+        """
+        if self.textbox.value == '' or not self.date_mapp:
+            return self.textbox.value
+        else:
+            return datetime.strptime(self.textbox.value, self.date_mapp)
+
+    def fill(self, value):
+        """Fill date to date box
+
+        Args:
+           value (datetime): datetime object.
+        Returns:
+            :py:class:`bool`
+        """
+        if isinstance(self.read(), datetime):
+            if value.date() == self.read().date():
+                return False
+
+        if not self.is_readonly:
+            date = datetime.strftime(value, self.date_mapp)
+            self.textbox.fill(date)
+            self.date_pick._elements[self.date_pick.active].click()
+            return True
+        else:
+            self.browser.click(self.textbox)
+            sw_value = self.date_pick.dp_switch.read().split(' ')
```

Comment:
I think you need to just set passed values w/o any parsing. Is there a reason to read those values and check them ?

Reply:
Yes, 
First switch value will be a combination of month and Year like `June 2018` so I am splitting it and check for `month` and `year` match or not. if not match it means I have to go for `year` or `month` changes. When we click first time to `switch` its come to month selection view. and its value is a year like `2018`.